python/lsst/daf/butler/cli/utils.py

```diff
@@ -398,9 +448,73 @@ def cli_handle_exception(func, *args, **kwargs):
         raise click.ClickException(msg.getvalue())
 
 
+class option_section:  # noqa: N801
+    """Decorator to add a section label between options in the help text of a
+    command.
+
+    Parameters
+    ----------
+    sectionText : `str`
+        The text to print in the section identifier.
+    """
+
+    def __init__(self, sectionText):
+        self.sectionText = "\n" + sectionText
+
+    def __call__(self, f):
+        # Generate a parameter declaration that will be unique for this
+        # section.
+        return click.option(f"--option-section-{str(uuid.uuid4())}",
+                            sectionText=self.sectionText,
+                            cls=OptionSection)(f)
+
+
+class MWPath(click.Path):
```

Comment:
When specifying a butler configuration file you can't use anything that assumes that it can open a local file. From the ctrl_mpexec PR I see that this is used for butler configs. Butler configs can be URIs that may or may not include the "butler.yaml" at the end (they can be `s3://` etc).  You could use `ButlerURI.exists()` to see if it exists but better to pass the string to `Butler()` directly without trying to check it.